src/richie/apps/core/context_processors.py

```diff
@@ -32,6 +33,12 @@ def site_metas(request):
             "domain": site_current.domain,
             "web_url": f"{protocol:s}://{site_current.domain:s}",
         },
+        "SOCIAL": {
+            "registration_endpoint": getattr(
+                settings, "SOCIAL_AUTH_EDX_OAUTH2_REGISTRATION_ENDPOINT", ""
+            )
+            or reverse("admin:index")
```

Comment:
Last parameter of `getattr` is the default value, so maybe write it like
```python
getattr(
    settings, "SOCIAL_AUTH_EDX_OAUTH2_REGISTRATION_ENDPOINT", reverse("admin:index")
)
```
to remove the `or` statement and simplify your code?

Reply:
This syntax is less elegant but it is more resilient. In fact, getattr return default value only if the property does not exist. In the case of this settings exists, it do not return the default value.

By using `or reverse(...)`, we prevent this case.